### ISSUE TITLE
`table` does not sanitize strings

### ISSUE THREAD

**Issue by umnikos** [nushell/nushell#12725]
### Describe the bug

Strings with special characters in them (newline, carriage return, etc.) should be shown differently if they're part of a table or otherwise shown as a string value instead of text to print. This applies to `table`, `grid`, and probably other commands as well.

### How to reproduce

```
~/De/test $ ls
 #   name   type   size    modified  
─────────────────────────────────────
 0   a      file    2 B   a year ago 
 1   b      file    2 B   a year ago 
 2   c      file    2 B   a year ago 
 3   d      file    2 B   a year ago 
 4   e      file    2 B   a year ago 
~/De/test $ touch (char newline)
~/De/test $ ls
 #   name   type   size    modified  
─────────────────────────────────────
 0          file    0 B   now        
                                     
 1   a      file    2 B   a year ago 
 2   b      file    2 B   a year ago 
 3   c      file    2 B   a year ago 
 4   d      file    2 B   a year ago 
 5   e      file    2 B   a year ago 
~/De/test $ 
```

### Expected behavior

Something similar to this:
```
~/De/test $ ls
 #   name   type   size    modified  
─────────────────────────────────────
 0   \n     file    0 B   now        
 1   a      file    2 B   a year ago 
 2   b      file    2 B   a year ago 
 3   c      file    2 B   a year ago 
 4   d      file    2 B   a year ago 
 5   e      file    2 B   a year ago 
~/De/test $ 
```

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.93.0                                          |
| major              | 0                                               |
| minor              | 93                                              |
| patch              | 0                                               |
| branch             |                                                 |
| commit_hash        |                                                 |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09)             |
| rust_channel       | stable-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26)             |
| build_time         | 2024-05-01 15:31:24 +03:00                      |
| build_rust_channel | release                                         |
| allocator          | mimalloc                                        |
| features           | default, sqlite, system-clipboard, trash, which |
| installed_plugins  |                                                 |

**Comment by devyn** [comment]
I think it's intentional that newlines are allowed inside tables, so that you can see multi-line values. We also allow ANSI escapes to go through and this is critical for some functionality - for example, see `ansi --list`'s `preview` column.

I don't think we're likely to change this because it's useful. If you absolutely need to see the escapes, you can use a format that shows you them, like `to json`

**Comment by umnikos** [comment]
I totally agree about the ansi colorcodes, and newlines can go either way, but there's other characters which would really benefit from escaping:
```
~/De/test $ ls
 #   name   type   size    modified  
─────────────────────────────────────
 0   a      file    2 B   a year ago 
 1   b      file    2 B   a year ago 
 2   c      file    2 B   a year ago 
 3   d      file    2 B   a year ago 
 4   e      file    2 B   a year ago 
~/De/test $ touch (ansi clear_entire_screen)
~/De/test $ ls
```
```






            file    0 B   now        
 1   a      file    2 B   a year ago 
 2   b      file    2 B   a year ago 
 3   c      file    2 B   a year ago 
 4   d      file    2 B   a year ago 
 5   e      file    2 B   a year ago 
~/De/test $ 
```

**Comment by devyn** [comment]
Hmm, but at that point we have to start parsing ANSI escape sequences to try to sort out the good ones from the bad ones. Is this coming up somewhere where it's really necessary?

**Comment by umnikos** [comment]
Not yet, but I'd argue it's a security risk, especially if the displayed table isn't `ls` but is instead some untrusted input.

Sorting out the good unicode from the bad unicode has already been done by standard utils and that work can be reused, so what's left is exceptions for newlines and ansi colors.

**Comment by devyn** [comment]
What's the difference between that and just running a command and getting that untrusted output as raw bytes? You can easily curl some web service and get some weird ANSI escapes doing things. I don't think there's an implication that the tables you see from nushell are sanitized in anyway.

It's not like this affects anything other than printing, because there aren't any strings that can magically do anything otherwise, and if you're really suspicious of some output for some reason, you'd probably want to run it through `debug` or `into binary`

**Comment by sholderbach** [comment]
I think while it would be tricky to figure out which escapes and general control characters should be filtered on the level of `table` it seems at first an alternative to consider escaping on the level of `ls`. 
But then the challenge would be a different one. We could either display the proper escaping as a quoted string you would need to paste manually or provide the faithful bytes if used in some `ls | ... | fs-op` pipeline.
